File: Selection sort/SelectionSort.java
import java.util.Arrays;

public class SelectionSort {
    public static void main(String[] args) {
        int[] array = {213, 235,21, 53, 24, 46};
        int max = 0;
        int pos = array.length-1;
        int checker = 0;
        while(pos>0) {
            for (int i = 1; i < array.length - checker; i++) {
                if (array[i] > array[max]) {
                    max = i;
                }
            }
            swap(array, max, pos);
            max = 0;
            pos--;
            checker++;
        }
        System.out.println(Arrays.toString(array));
    }


    static void swap(int[] array, int max, int position){
        int temp = array[position];
        array[position] = array[max];
        array[max] = temp;
    }
}

File: C/SelectionSort/Selectionsort.c
#include <stdio.h>

void swap(int arr[5], int max, int last){
    int temp = arr[max];
    arr[max] = arr[last];
    arr[last] = temp;
}



void main(){
    int array[5] = {5, 4, 3 ,2, 1};
    int len = sizeof(array)/sizeof(int);

    for(int i = 0; i < len - 1; i++){
        int max = 0;
        for(int j = 1; j < len - i; j++){
            if(array[j] > array[max]){
                max = j;        
            }
        }
        swap(array, max, len - i - 1);
    }

    for(int i = 0; i<len; i++){
        printf(" %d ", array[i]);
    }
}